### PR TITLE
AF-238 orderable attribute

### PR DESCRIPTION
**Pull request by sandlerr**
:v:

/cc @zendesk/vegemite

### Description

Add an orderable attribute to the locations. IMO this is a great candidate for replacement with a [Struct](http://ruby-doc.org/core-2.3.0/Struct.html) but they are a bit less flexible and people are less familiar with them.

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-238

### Risks
* malformed installed.js (apps don't load)
* breaks reordering

### CODE REVIEW

**Comment by adammw** [comment]
I'd be happy with a struct too, but then you're essentially building a Location class in ZAS. Not that it's a bad thing, just then we'd have duplication with ZAM.

**Comment by sandlerr** [comment]
@adammw not sure I understand how using a struct would increase duplication

**Comment by danielbreves** [comment]
We'll need to make sure we do a minor bump after merging, since it's a breaking change.

**Comment by danielbreves** [comment]
:+1:

File: lib/zendesk_apps_support/location.rb [hunk 1]
```diff
@@ -3,16 +3,16 @@ module Location
...
+        'nav_bar' => { 'id' => 2, 'orderable' => true },
+        'ticket_sidebar' => { 'id' => 3, 'orderable' => true },
+        'new_ticket_sidebar' => { 'id' => 4, 'orderable' => true },
+        'user_sidebar' => { 'id' => 5, 'orderable' => true },
+        'organization_sidebar' => { 'id' => 6, 'orderable' => true },
+        'background' => { 'id' => 7, 'orderable' => false }
       },
       'zopim' => {
-        'chat_sidebar' => 8
+        'chat_sidebar' => { 'id' => 8, 'orderable' => false }
```
Comment (adammw) [inline lib/zendesk_apps_support/location.rb]: why is chat sidebar not reorderable?

**Comment by adammw** [comment]
I'm still not that convinced that it belongs in ZAS if nothing in ZAS (or ZAT) uses it...

File: lib/zendesk_apps_support/location.rb (hunk 1 above)
Comment (sandlerr) [inline lib/zendesk_apps_support/location.rb]: it only supports one app

Comment (adammw) [inline lib/zendesk_apps_support/location.rb]: the code doesn't know that. when we have multiple zopim apps we'd have to come back and update this... seems like a waste if we already have host-level filtering

**Comment by adammw** [comment]
:+1:

**Pull request merged**